riscv/processor.cc

```diff
@@ -215,7 +215,7 @@ void processor_t::parse_isa_string(const char* str)
   else
     bad_isa_string(str, "Spike supports either RV32I or RV64I");
   if (isa_string[4] == 'g')
-    isa_string = isa_string.substr(0, 4) + "imafd" + isa_string.substr(5);
+    isa_string = isa_string.substr(0, 4) + "imafd" + isa_string.substr(5) + "_zicsr_zifencei";
```

Comment:
If the ISA string already has Zicsr or Zifencei, this would add it again...is that okay?

Reply:
It's not a problem now, because we don't enforce canonical order of multi-letter extension names, and we allow repetition.  This is arguably a feature, because it makes it easier to assemble ISA strings piecemeal in scripts that invoke Spike.

However, I can see it becoming a problem eventually, and since this has no functional benefit, I think I've changed my mind.  Instead, I'll leave a comment saying that G also implies Zicsr + Zifencei, but since we include those implicitly for backwards compatibility, we don't need to explicitly list them here.